### PR TITLE
[FIX] account_edi_proxy_client, l10n_it_edi, l10n_it_edi_sdicoop: Fixing Base64 issues

### PR DESCRIPTION
**Pull request by lordkrandel**
When encoding a string or bytes with base64, the resulting byte string must be decoded to get the string back.

* `addons/account_edi_proxy_client/models/account_edi_proxy_user.py`
  AccountEdiProxyUser._register_proxy_user()

* `addons/l10n_it_edi/models/account_invoice.py`
  AccountMove._prepare_fatturapa_export_values()

* `addons/l10n_it_edi_sdicoop/models/account_edi_format.py`
  AccountEdiFormat._l10n_it_post_invoices_step_1()

### CODE REVIEW

File: addons/account_edi_proxy_client/models/account_edi_proxy_user.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 _logger = logging.getLogger(__name__)
 
 
+# IAP Server address.
+# Customize this address if you want to use the IAP Test Server
 SERVER_URL = 'https://l10n-it-edi.api.odoo.com'
```
Comment (lordkrandel) [inline addons/account_edi_proxy_client/models/account_edi_proxy_user.py:22]: @jco-odoo Do you think that this should also be a parameter? 
We cannot use test mode if we don't modify this line, so we would just have tests from local instances with that line  modified.

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ def _l10n_it_post_invoices_step_1(self, invoices):
                 'name': filename,
                 'res_id': invoice.id,
                 'res_model': invoice._name,
-                'datas': base64.encodebytes(xml),
+                'datas': base64.b64encode(xml.encode()),
```
Comment (jco-odoo) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:119]: we can also use the raw field here.  (where we don't do b64encode)

File: addons/account_edi_proxy_client/models/account_edi_proxy_user.py (hunk 1 above)
Comment (jco-odoo) [reply addons/account_edi_proxy_client/models/account_edi_proxy_user.py:22]: Or a test env config param and the url hard coded?  In any case, this is not some demo environment, but an experimentation environment, which will mostly depend on people having access locally.  But no, I think your suggestion makes sense.  (but if the system param does not exist, you use this as standard)

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py (hunk 2 above)
Comment (lordkrandel) [reply addons/l10n_it_edi_sdicoop/models/account_edi_format.py:119]: 👍🏻

File: addons/account_edi_proxy_client/data/config_demo.xml [hunk 3]
```diff
@@ -5,5 +5,9 @@
         <field name="key">account_edi_proxy_client.demo</field>
         <field name="value">True</field>
     </record>
+    <record id="account_edi_proxy_client_edi_server_url" model="ir.config_parameter">
+        <field name="key">account_edi_proxy_client.edi_server_url</field>
+        <field name="value">https://l10n-it-edi.api.odoo.com</field>
+    </record>
```
Comment (lordkrandel) [inline addons/account_edi_proxy_client/data/config_demo.xml:11]: To be removed.

File: addons/account_edi_proxy_client/models/res_config_settings.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import fields, models
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import DEFAULT_SERVER_URL
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    account_edi_proxy_client_edi_server_url = fields.Char(
+        string='EDI Server Url',
+        config_parameter='account_edi_proxy_client.edi_server_url',
+        default=DEFAULT_SERVER_URL)
```
Comment (lordkrandel) [inline addons/account_edi_proxy_client/models/res_config_settings.py:13]: To be removed.

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 5]
```diff
@@ -116,7 +117,7 @@ def _l10n_it_post_invoices_step_1(self, invoices):
                 'name': filename,
                 'res_id': invoice.id,
                 'res_model': invoice._name,
-                'datas': base64.b64encode(xml.encode()),
+                'raw': xml.encode(),
```
Comment (lordkrandel) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:120]: Split in two commits

**Pull request closed without merging**